lib/cocoapods/target/build_settings.rb

```diff
@@ -800,7 +800,8 @@ def linker_names_from_libraries(libraries)
                                     uniq
             static_libraries_to_import.concat linker_names_from_libraries(xcframework_libraries)
           end
-          static_libraries_to_import << target.product_basename if target.should_build? && target.build_as_static_library?
+          testsHostedInApp = test_xcconfig && !target.test_app_hosts_by_spec[non_library_spec].nil?
```

Comment:
nit use snake case for variable name `lib_hosted_in_app`

Reply:
Sorry about that. I am fairly inexperienced in ruby.

Reply:
no worries at all!

Can you also add a spec (test) in `build_settings_spec.rb`?

Reply:
I looked into the existing tests in `build_settings_spec.rb` and I am not sure what test to add over there. Do you have any particular scenario you have in mind?

Reply:
@dnkoutso 
Hey, I don't know if you had the chance to see my last reply.

I spent some time looking into the existing `build_settings_spec.rb` and tried writing some myself. The test was testing the same thing I already added a tested in `pod_target_settings_spec.rb` so I was worried I was doing something wrong.

The test I wrote was something like:
```ruby
it 'does not add OTHER_SWIFT_FLAGS to the xcconfig if the target does not use swift' do
  target = fixture_pod_target('...')
  build_settings = pod(target)
  other_ld_flags = build_settings.xcconfig.to_hash['OTHER_LDFLAGS']
  other_swift_flags.should == '$(inherited) -ObjC'
end
```

Am I missing something here? Is there another test that can be written?

Reply:
@dnkoutso 
Hey, 
I just added a spec containing 2 tests in `build_settings_spec.rb`. In these tests, instead of initializing a `PodTarget` with `fixture_pod_target` I stubbed most of the functionality. I hope it makes sense :).

Reply:
@dnkoutso  I am not sure what's the status of this change. Should someone review it? Should I do something else?